libs/azure-storage/langchain_azure_storage/document_loaders.py

```diff
@@ -265,14 +269,18 @@ async def _ayield_blob_names(
             for blob_name in self._blob_names:
                 yield blob_name
         else:
-            async for blob_name in async_container_client.list_blob_names(
-                name_starts_with=self._prefix
+            async for blob in async_container_client.list_blobs(
+                name_starts_with=self._prefix, include="metadata"
             ):
-                yield blob_name
+                if not self._is_adls_directory(blob):
+                    yield blob.name
 
     def _get_default_document(
         self, blob_content: bytes, blob_client: Union[BlobClient, AsyncBlobClient]
     ) -> Document:
         return Document(
             blob_content.decode("utf-8"), metadata={"source": blob_client.url}
         )
+
+    def _is_adls_directory(self, blob: BlobProperties) -> bool:
+        return blob.size == 0 and blob.metadata.get("hdi_isfolder") == "true"
```

Comment:
In between the `blob.size` and the `hdi_isfolder` check, let's add a `blob.metadata`. So:
```python
blob.size == 0 and blob.metadata and blob.metadata.get("hdi_isfolder") == "true"
```
Mainly blobs could be empty and not be directories and have `metadata` that equates to `None`. This additional clause will make sure we short-circuit early to avoid trying to call `.get()` on a `None` value and cause the entire listing to error out.